UnitTests/UnitTests.UWP/UI/Controls/Test_RangeSelector.cs

```diff
@@ -0,0 +1,343 @@
+
+        [DataRow(   1,  100,  -90,  -10,    0,                 1,    0,    0,    0, 0.01, DisplayName = "Minimum  > Maximum, Minimum  > RangeStart  < RangeEnd  < Maximum")]
+        [DataRow(   1,  100,  -10,  -90,    0,                 1,    0,    0,    0, 0.01, DisplayName = "Minimum  > Maximum, Minimum  > RangeStart  > RangeEnd  < Maximum")]
+        [DataRow(   1,  100,  -50,  -50,    0,                 1,    0,    0,    0, 0.01, DisplayName = "Minimum  > Maximum, Minimum  > RangeStart == RangeEnd  < Maximum")]
+
+        [DataRow(   1,  100,  110,  190,    0,                 1,    0, 0.01, 0.01, 0.01, DisplayName = "Minimum  > Maximum, Minimum  < RangeStart  < RangeEnd  > Maximum")]
+        [DataRow(   1,  100,  190,  110,    0,                 1,    0, 0.01, 0.01, 0.01, DisplayName = "Minimum  > Maximum, Minimum  < RangeStart  > RangeEnd  > Maximum")]
+        [DataRow(   1,  100,  150,  150,    0,                 1,    0, 0.01, 0.01, 0.01, DisplayName = "Minimum  > Maximum, Minimum  < RangeStart == RangeEnd  > Maximum")]
+#pragma warning restore SA1025, SA1008
+        public async Task Initialize(double stepFrequency, double minimum, double rangeStart, double rangeEnd, double maximum, double expectedStepFrequency, double expectedMinimum, double expectedRangeStart, double expectedRangeEnd, double expectedMaximum)
+        {
+            var input = new TestRecord(stepFrequency, minimum, rangeStart, rangeEnd, maximum);
+            var expected = new TestRecord(expectedStepFrequency, expectedMinimum, expectedRangeStart, expectedRangeEnd, expectedMaximum);
+
+            await App.DispatcherQueue.EnqueueAsync(async () =>
+            {
+                var r = BuildRangeSelecor(input);
+
+                await SetTestContentAsync(r);
+
+                var actual = BuildTestRecord(r);
+
+                Assert.AreEqual(expected, actual);
+            });
+        }
+
+        [TestMethod]
+        [TestCategory("Set Prop")]
+
+        // Given:Sf   Min   Max   Start End      Set:Minimum       Then:Sf  Min     Max     Start   End
+#pragma warning disable SA1008, SA1025
+        [DataRow(   1,    0,   10,   90,  100,             0,             1,    0   ,  10   ,  90   , 100   )]
+        [DataRow(   1,    0,   10,   90,  100,           -10,             1,  -10   ,  10   ,  90   , 100   )]
+        [DataRow(   1,    0,   10,   90,  100,            10,             1,   10   ,  10   ,  90   , 100   )]
+        [DataRow(   1,    0,   10,   90,  100,            50,             1,   50   ,  50   ,  90   , 100   )]
+        [DataRow(   1,    0,   10,   90,  100,            90,             1,   90   ,  90   ,  90   , 100   )]
+        [DataRow(   1,    0,   10,   90,  100,           100,             1,  100   , 100   , 100   , 100   )]
+        [DataRow(   1,    0,   10,   90,  100,           110,             1,  110   , 110.01, 110.01, 110.01)]
```

Comment:
If the first 5 values are always the same, is there a point in providing them everytime or would it be better to scratch them as parameter for easier readability?

Reply:
Good point, took that idea and removed a lot of the repetition from the other tests also, by splitting them up more. I think its more readable overall, lmk what you think.

Reply:
I think it really helps a lot, it's easier to read why those testcases are there since you can focus on the values instead of reading multiple values that are not actually important to the test.